plugins/vpp/ifplugin/nat_config.go

```diff
@@ -1068,6 +1104,37 @@ func (plugin *NatConfigurator) diffIdentity(oldMappings, newMappings []*nat.Nat4
 	return
 }
 
+// diffVirtualReassembly compares virtual reassembly from old config, returns virtual reassembly to be configured, or nil
+// if no changes are needed
+func isVirtualReassModified(oldReass, newReass *nat.Nat44Global_VirtualReassembly) *nat.Nat44Global_VirtualReassembly {
+	// If both are nil, return nil
+	if oldReass == nil && newReass == nil {
+		return nil
+	}
+	// If reassembly was not set before, configure new one
+	if oldReass == nil && newReass != nil {
+		return newReass
+	}
+	// If reassembly config was removed, set to default
+	if oldReass != nil && newReass == nil {
+		return getDefaultVr()
+	}
+	// Compare configs and return new one if there is a change, nil otherwise
+	if oldReass.DropFrag != newReass.DropFrag {
```

Comment:
You could just compare whole struct: `if *oldReass != *newReass {`.

Reply:
Then you could combine it easily with second if case to: `if newReass != nil && (oldReass == nil || *oldReass != *newReass) {`

Reply:
And then the first case seems useless.